### PR TITLE
[xtext generator] upgraded 'GeneratedJavaFileAccess' to expect the 'typeComment' as a 'StringConcatenationClient'…

### PR DESCRIPTION
**Pull request by sailingKieler**
… and process imports of referenced classes. The generated Java files affected by this change are amended.

Signed-off-by: Christian Schneider <christian.schneider@itemis.de>

### CODE REVIEW

File: plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/GeneratedJavaFileAccess.xtend [hunk 1]
```diff
@@ -25,9 +26,17 @@ class GeneratedJavaFileAccess extends JavaFileAccess {
 		if (!markedAsGenerated)
 			throw new IllegalArgumentException("It's always generated");
 	}
-
+	
+	def setTypeComment(StringConcatenationClient javaContent) {
```
Comment (sailingKieler) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/GeneratedJavaFileAccess.xtend]: A question on the return type that is now 'CharSequence'. Is that fine or should I explicitly set it to 'void'?
(did it the same way like in 'JavaFileAccess.setContent(...)')

Comment (szarnekow) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/GeneratedJavaFileAccess.xtend]: I think setters should return void in general. Also in JavaFileAccess. Is the return value of JavaFileAccess.content used somewhere?

Comment (sailingKieler) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/GeneratedJavaFileAccess.xtend]: the setter stem's from 'TextFileAccess' and declaring it explicitly 'void' does not break anything.
Personally I'm not a fan of void, but rather would return the '...FileAccess' instance itself, like it is done in ``GuiceModuleAccess.BindingFactory``. But that may be a bit overdressed here.
Anyway, returning the 'CharSequence' doesn't make sense to me.

Comment (szarnekow) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/GeneratedJavaFileAccess.xtend]: Please let the setters return 'void'.

Comment (sailingKieler) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/GeneratedJavaFileAccess.xtend]: done for both 'setContent(...)' and 'setTypeComment(...)'.

**Comment by szarnekow** [comment]
Jenkins test

**Pull request merged**